tree/treeplayer/src/TBranchProxy.cxx

```diff
@@ -18,13 +18,31 @@ of the autoloading of branches as well as all the generic setup routine.
 #include "TLeaf.h"
 #include "TBranchElement.h"
 #include "TBranchObject.h"
+#include "TCollection.h" // TRangeStaticCast
 #include "TStreamerElement.h"
 #include "TStreamerInfo.h"
+#include "ROOT/InternalTreeUtils.hxx" // GetFileNamesFromTree, GetTreeFullPaths
+
+#include <string>
+#include <string_view>
 
 ClassImp(ROOT::Detail::TBranchProxy);
 
 using namespace ROOT::Internal;
 
+namespace {
+/**
+ * \brief Find if the input branch name is the prefix for other sub-branches in the same tree.
+ */
+bool AreThereSubBranches(std::string_view parentBrName, TTree &tree)
+{
+   for (auto *leaf : ROOT::Detail::TRangeStaticCast<TLeaf>(tree.GetListOfLeaves()))
+      if (std::string(leaf->GetName()).rfind(std::string(parentBrName) + ".", 0) != std::string::npos)
```

Comment:
Are we sure that `parentBrName` never ends with a dot?

(Indepedently) Why this version of the test rather than:
```suggestion
      if (std::string(leaf->GetName()).find(std::string(parentBrName) + ".") == 0)
```